### PR TITLE
Ssh connection

### PR DESCRIPTION
**Pull request by LukasFehring**
Add possibility of using a ssh jumphost to conenct to a mysql-server that only allows connections from specific ips

## Description
- Adapt credentials file to yaml
- Add ssh section to credentials file
- add flag `use_shh=False` to the pyEpxerimenter. If using mysql and the parameter is set to true, the ssh jumphost will be used

## Motivation and Context

One can only connect to our databases from the institutes internet. We want to allow connections from clusters

#### Type Of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How has This Been Tested?

- Database provider: mysql
- Python version: 3.10
- Operating System: Any

Check whether previously working tests still work.
Create one additionl test using ssh. Note that one can not simply use all testcases with ssh, because after closing an ssh connection in python it is not instantly closed. This leads to issues when trying to build a new ssh connection

Add [sshtunnel](https://pypi.org/project/sshtunnel/) libary 

## Does this Close/Impact Existing Issues?

None

## Checklist

- [x] My change is based on the latest stage of the `develop` branch.
- [x] My change required a change of the documentation, which has been done.
- [x] I checked that the documentation can be build, visualizes everything as expected, and does not contain any warnings.
- [x] I have added/adapted tests to cover my changes.
- [x] The tests can be executed successfully.
- [x] The notebooks can be executed successfully.
- [x] The notebooks can be executed with `mysql` as provider.
- [x] I have added a description of the changes to `CHANGELOG.rst`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -137,14 +137,21 @@ dmypy.json
...
+config/example_general_usage.yml
+config/example_logtables.yml
+config/example_pause_and_continue.yml
 output/
 
 # codecarbon
 .codecarbon.config
 emissions.csv
-config/*.yml
+config/.yml
```
Comment (tornede) [inline .gitignore:152]: why is "*" removed?

File: .gitignore [hunk 2]
```diff
@@ -137,14 +137,21 @@ dmypy.json
...
 
 # development folder
-development/
\ No newline at end of file
+development/
+
```
Comment (tornede) [inline .gitignore:156]: remove everything but one empty line

File: CHANGELOG.rst [hunk 3]
```diff
@@ -7,7 +7,8 @@ v1.4.0 (??.??.2024)
 
 Feature
 -------
-- Changed supported experiment configuration file type to YAML. 
+- Changed supported experiment configuration file type to YAML.
+- Add support for ssh jump hosts in the database connection.
```
Comment (tornede) [inline CHANGELOG.rst:11]: Additionally: 
- Changed supported database configuration file type to YAML.

File: docs/source/usage/execution.rst [hunk 4]
```diff
@@ -32,14 +32,14 @@ Additionally, further information can be given to ``PyExperimenter``:
 
 - ``experiment_configuration_file_path``: The path of the :ref:`experiment configuration file <experiment_configuration_file>`. Default: ``config/experiment_configuration.cfg``.
 - ``database_credential_file_path``: The path of the :ref:`database credential file <database_credential_file>`. Default: ``config/database_credentials.cfg``
+- ``use_ssh_tunnel``: Specifies if a SSH tunnel will be used to connect to the database. Default: ``False``. If ``use_ssh_tunnel`` is set to ``True``, creating a ``PyExperimenter`` will also open an ssh tunnel, which should be closed manually. For more information check :ref:`close_ssh_tunnel <close_ssh_tunnel>`.
```
Comment (tornede) [inline docs/source/usage/execution.rst:35]: Reference to the database config doc, as it requires a specific ssh part

File: docs/source/usage/execution.rst [hunk 5]
```diff
@@ -164,7 +164,7 @@ Tracking information about the carbon footprint of experiments is supported via
 Pausing and Unpausing Experiments
 ---------------------------------
 
-For convenience, we support pausing and unpausing experiments. This means that you can use one ``PyExperimenter`` to start an experiment, which will be paused after certain operations. Therefore, it can be resumed later on. Afterwards, depending on the parametrization of ``execute()`` of the ``PyExperimenter`` instance (see :ref:`asdf <execute_experiments:>`), the experimenter terminates or another experiment will be started. 
+For convenience, we support pausing and unpausing experiments. This means that you can use one ``PyExperimenter`` to start an experiment, which will be paused after certain operations. Therefore, it can be resumed later on. Afterwards, depending on the parametrization of ``execute()`` of the ``PyExperimenter`` instance (see :ref:`asdf <execute_experiments>`), the experimenter terminates or another experiment will be started.
```
Comment (tornede) [inline docs/source/usage/execution.rst:167]: the "asdf" should not have been permanent :D Please replace accordingly

File: docs/source/usage/execution.rst [hunk 6]
```diff
@@ -201,3 +201,16 @@ At a later point in time, the experiment can be unpaused and continued. This can
...
 
+
+
+.. _close_ssh_tunnel:
+
+----------------
+Close SSH Tunnel
+----------------
+
+If an SSH tunnel was opened during the creation of the ``PyExperimenter``, it has to be closed manually by calling
```
Comment (tornede) [inline docs/source/usage/execution.rst:212]: .. by calling it after the execution of PyExperimenter terminates

**Review by tornede — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (LukasFehring) [reply .gitignore:152]: Youre correcht, this should be removed as well

File: CHANGELOG.rst (hunk 3 above)
Comment (LukasFehring) [reply CHANGELOG.rst:11]: I adapted this, but different then mentioned by you

File: docs/source/usage/execution.rst (hunk 4 above)
Comment (LukasFehring) [reply docs/source/usage/execution.rst:35]: I do not understand

**Review by tornede — APPROVED** [review]

**Pull request merged**